File: general-server/src/main/java/com/song/middleware/server/domain/ServiceDO.java
package com.song.middleware.server.domain;

import java.io.Serializable;
import lombok.Getter;
import lombok.Setter;

/**
 * Created by song on 2017/8/5.
 */
@Getter
@Setter
public class ServiceDO implements Serializable {

    private static final long serialVersionUID = 7208074479190445962L;

    private long id;

    private String name;

    private String description;

    private String ip;

    private String port;

    /**
     * 服务当前所处的状态，默认为down
     */
    private int status = ServiceStatus.DOWN.getCode();

    /**
     * 健康检查链接
     */
    private String healthCheckUrl;

    /**
     * 健康检查间隔
     * 单位是秒
     */
    private long healthCheckInterval;
}

File: general-server/src/main/java/com/song/middleware/server/domain/ServiceDO.kt
package com.song.middleware.server.domain

import java.io.Serializable

/**
 * Created by song on 2017/8/5.
 */
class ServiceDO : Serializable {

    var id: Long = 0

    var name: String? = null

    var description: String? = null

    var ip: String? = null

    var port: String? = null

    /**
     * 服务当前所处的状态，默认为down
     */
    var status = ServiceStatus.DOWN.code

    /**
     * 健康检查链接
     */
    var healthCheckUrl: String? = null

    /**
     * 健康检查间隔
     * 单位是秒
     */
    var healthCheckInterval: Long = 0

    companion object {

        private const val serialVersionUID = 7208074479190445962L
    }
}
